test/integration/fake_upstream.h

```diff
@@ -150,7 +163,7 @@ class FakeStream : public Http::RequestDecoder,
   ABSL_MUST_USE_RESULT testing::AssertionResult
   waitForGrpcMessage(Event::Dispatcher& client_dispatcher, T& message,
                      std::chrono::milliseconds timeout = TestUtility::DefaultTimeout) {
-    auto end_time = timeSystem().monotonicTime() + timeout;
+    auto end_time = timeSystem().realMonotonicTimeDoNotUseWithoutScrutiny() + timeout;
```

Comment:
could we capture this pattern in class like:
```
class RealTimeBounds {
 public:
  RealTimeBounds(Duration limit)
   : end_time_(timeSystem().realMonotonicTimeDoNotUseWithoutScrutiny() + limit) {}

 bool canContinue() const { return timeSystem().realMonotonicTimeDoNotUseWithoutScrutiny() < end_time_; }
};
```

(not quite right but close enough for a github comment).

That pattern is used quite a bit and it might simplify code.

Just a thought: if this is all going to go away (because we someday will not need these real-time delays) then maybe this is not worth it.

Reply:
Yes good idea let me do that.